src/parsy/__init__.py

```diff
@@ -538,6 +545,27 @@ def string_parser(stream, index):
     return string_parser
 
 
+def bstring(expected_bytes: bytes, transform: Callable[[bytes], bytes] = noop) -> Parser:
```

Comment:
why have this as a separate function? this code is a direct copy of `parsy.string()`, which already works just fine on binary strings (`bytes`):

``` python
>>> p = parsy.string(b'abc')
>>> p.parse_partial(b'abcdef')
(b'abc', b'def')
```

… except for its type annotation which contains `str` (a few times). changing those annotations to [`typing.AnyStr`](https://docs.python.org/3/library/typing.html#typing.AnyStr) should be enough.